data_structures/binary_tree/left_view_of_binary_tree.py

```diff
@@ -0,0 +1,31 @@
+class Node:
+    def __init__(self, data, left=None, right=None):
+        self.data = data
+        self.left = left
+        self.right = right
+
+
+def fun(root, d, l):
```

Comment:
We need a better name for this function.  For me `fun` is rolling around on the floor with my dog.  Every reader of your code will have interesting ideas about what fun is but few will think of `data_and_level()` which describes what this function does.  As discussed in `CONTRIBUTING.md`, we encourage contributors to use self-documenting function names and variable names.  Your code should describe itself to your reader.  Single-letter variable names like `d` are seriously uncool.  They make the code look like it was written in the 1970’s.
```suggestion
def data_and_level(root: Node, data_and_level_info: dict, level: int) -> None:
```